agent/dns.go

```diff
@@ -673,6 +682,40 @@ RPC:
 	}
 }
 
+// addNSAndARecordsForDomain uses the agent's advertise address to
+func (d *DNSServer) addNSAndARecordsForDomain(msg *dns.Msg) {
+	serverAddrs := d.agent.delegate.ServerAddrs()
+	for _, addr := range serverAddrs {
+		ipAddrStr := strings.Split(addr, ":")[0]
+		nsName := "ns." + ipAddrStr + "." + d.domain
+		ip := net.ParseIP(ipAddrStr)
+		if ip != nil {
+			ns := &dns.NS{
+				Hdr: dns.RR_Header{
+					Name:   d.domain,
+					Rrtype: dns.TypeNS,
+					Class:  dns.ClassINET,
+					Ttl:    0,
```

Comment:
open question around TTL for this part. Should it be > 0? 

Note that the A record below with the ip address mapped to the ns record name does set a TTL